yt/utilities/on_demand_imports.py

```diff
@@ -665,3 +665,26 @@ def __getattr__(self, attr):
 
 
 _pandas = pandas_imports()
+
+class firefly_imports:
```

Comment:
I think this class is missing a `__init__` method. Additionally, I don't think the implemented properties mocking submodules are necessary, see how the `pandas_import` class above for a minimalist working implementation.

Reply:
```
class firefly_imports:
    _name = "firefly"
    _module = None

    def __init__(self):
        try:
            import Firefly as myself

            self._module = myself
        except ImportError:
            self._module = NotAModule(self._name)

    def __getattr__(self,attr):
        return getattr(self._module,attr)


_firefly = firefly_imports()
```

Does not work and raises an attribute error for the sub-module `data_reader`. I based my original implementation on the [yaml imports](https://github.com/yt-project/yt/blob/main/yt/utilities/on_demand_imports.py#L554) which does not raise an attribute error.